### ISSUE TITLE
Async Pipe for map of Observables

### ISSUE THREAD

**Issue by Jack-Bachman** [angular/angular#55693]
### Which @angular/* package(s) are relevant/related to the feature request?

common

### Description

I work within a large codebase and commonly see this pattern when using multiple observables. Often, developers map observables to fields within a mapping and then name that mapping. This leads to blocks that look like the following:
```
<div *ngIf="{ 
    a: aObservable$ | async,
    b: bObservable$ | async,
    c: cObservable$ | async
} as obs">
  <p>{{ obs.a }}</p>
  <p>{{ obs.b }}</p>
  <p>{{ obs.c }}</p>
</div>
```
There is an annoying bug with this code though. When one of the observables within the mapping returns a value, the result is defined, therefore the entire NgIf results to `true` (since an object is truthy). This causes race conditions and cases where the nested html is rendered with null fields instead of the expected emitted values.

The proper way to fix this issue is to break it into multiple async calls:
```
<div *ngIf="aObservable$ | async as a">
  <div *ngIf="bObservable$ | async as b">
    <div *ngIf="cObservable$ | async as c">
      <p>{{ a }}</p>
      <p>{{ b }}</p>
      <p>{{ c }}</p>
    </div>
  </div>
</div>
```
This leads to deep nesting though and is cumbersome to write and consume.

Within our codebase I have implemented the proposed solution and was wondering if others were open to me contributing it to @angular/common pipes.

### Proposed solution

I implemented the `multiasync` pipe. It works just like the `async` pipe but operates on a mapping to observables instead of a singular observable value.
```
<div *ngIf="{ 
    a: aObservable$,
    b: bObservable$,
    c: cObservable$
} | multiasync as obs">
  <p>{{ obs.a }}</p>
  <p>{{ obs.b }}</p>
  <p>{{ obs.c }}</p>
</div>
```

Internally, `multiasync`is similar to async but instead of a singular subscription it is subscribing to all the member observables using `combineLatest`. This results in a value once all the observables emit. It will also re-render the inner html when any of observables update after that initial emit.

### Alternatives considered

I tried to implement this with multiple async pipes but nothing worked as intended or was ugly to write and hard to use. I think the above pattern is fairly common (I could be wrong), and if so, this might be a nice ergonomic addition to angular's pipes.

**Comment by JeanMeche** [comment]
It looks to me that this issue can mostly be addressed when we'll have local variables, see #15280

**Comment by alxhub** [comment]
Yep, closing as a spiritual duplicate of #15280.
